api/krusty/remoteload_test.go

```diff
@@ -41,103 +28,254 @@ spec:
+		"github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6")
+	assert.NoError(err)
+	checkYaml(m, multibaseDevExampleBuild, assert)
+}
+
+func TestRemoteResourceHttps(t *testing.T) {
+	tests := map[string]remoteResourceCase{
+		"basic": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`,
+		},
+		".git repo suffix, no slash suffix": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize.git//examples/multibases/dev?ref=v1.0.6`,
+		},
+		"repo": {
+			kustomization: `
+resources:
+- https://github.com/annasong20/kustomize-test.git?ref=main`,
+		},
+		"raw remote file": {
+			kustomization: `
+resources:
+- https://raw.githubusercontent.com/kubernetes-sigs/kustomize/v3.1.0/examples/multibases/base/pod.yaml
+namePrefix: dev-`,
+		},
+	}
+
+	for name, test := range tests {
+		test := test
+		if !test.skip {
+			t.Run(name, func(t *testing.T) {
+				testRemoteResource(assert.New(t), test.kustomization, multibaseDevExampleBuild)
+			})
+		}
 	}
-	assert.NoError(t, err)
-	yml, err := m.AsYaml()
-	assert.NoError(t, err)
-	assert.Equal(t, `apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    app: myapp
-  name: dev-myapp-pod
-spec:
-  containers:
-  - image: nginx:1.7.9
-    name: nginx
-`, string(yml))
-	assert.NoError(t, fSys.RemoveAll(tmpDir.String()))
 }
 
-func TestRemoteResourceGitHTTP(t *testing.T) {
-	output := `apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    app: myapp
-  name: dev-myapp-pod
-spec:
-  containers:
-  - image: nginx:1.7.9
-    name: nginx
-`
-	tests := []struct {
-		input []byte
-	}{
-		{
-			input: []byte(`
+func TestRemoteResourceSsh(t *testing.T) {
+	tests := map[string]remoteResourceCase{
+		"scp shorthand": {
+			kustomization: `
 resources:
-- https://github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6
-`),
+- git@github.com:kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`,
 		},
-		{
-			input: []byte(`
+		"full ssh, no ending slash": {
+			kustomization: `
 resources:
-- https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6
-`),
+- ssh://git@github.com/kubernetes-sigs/kustomize//examples/multibases/dev?ref=v1.0.6`,
 		},
-		{
-			input: []byte(`
+		"repo": {
+			kustomization: `
 resources:
-- git::https://github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6
-`),
+- ssh://git@github.com/annasong20/kustomize-test.git?ref=main`,
 		},
-		{
-			input: []byte(`
+	}
+
+	for name, test := range tests {
+		test := test
+		if !test.skip {
+			t.Run(name, func(t *testing.T) {
+				testRemoteResource(assert.New(t), test.kustomization, multibaseDevExampleBuild)
+			})
+		}
+	}
+}
+
+func TestRemoteResourcePort(t *testing.T) {
+	tests := map[string]remoteResourceCase{
+		"https": {
+			skip: true, // timeout
+			kustomization: `
+resources:
+- https://github.com:443/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`,
```

Comment:
For tests that consistently fail both locally and CI, we should have the `remoteResourceCase` struct contain an `expectedErr` field. Rather than skip the test, compare the actual error message with the expected error message, and leave a comment here that we may eventually want to change this behavior.

Reply:
Does this also apply for `TestRemoteResourceParameters`? The test is currently skipped.

Reply:
Yup!